cloudformation/template.yaml

```diff
@@ -0,0 +1,684 @@
+          Statement:
+          - Action:
+            - aws-marketplace:BatchMeterUsage
+            - aws-marketplace:ResolveCustomer
+            Effect: Allow
+            Resource: '*'
+
+  LambdaApiGatewayExecutionPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !GetAtt DevPortalLambdaFunction.Arn
+      Principal: apigateway.amazonaws.com
+      SourceArn: !Join
+        - ''
+        - - 'arn:aws:execute-api:'
+          - !Ref 'AWS::Region'
+          - ':'
+          - !Ref 'AWS::AccountId'
+          - ':'
+          - !Ref ApiGatewayApi
+          - '/*/*'
+
+  LambdaCognitoUserPoolExecutionPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !GetAtt PreSignupLambdaFunction.Arn
+      Principal: cognito-idp.amazonaws.com
+      SourceArn: !Join
+        - ''
+        - - 'arn:aws:cognito-idp:'
+          - !Ref 'AWS::Region'
+          - ':'
+          - !Ref 'AWS::AccountId'
+          - ':userpool/'
+          - !Ref CognitoUserPool
+
+  LambdaSNSExecutionPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !GetAtt SubscriptionListenerLambdaFunction.Arn
+      Principal: sns.amazonaws.com
+      SourceArn: !Join
+        - ''
+        - - 'arn:aws:sns:us-east-1:287250355862:aws-mp-subscription-notification-'
+          - !Ref MarketplaceSubscriptionTopicName
+
+  LambdaSNSExecutionTestPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: lambda:InvokeFunction
+      FunctionName: !GetAtt SubscriptionListenerLambdaFunction.Arn
+      Principal: sns.amazonaws.com
+      SourceArn: !Join
+        - ''
+        - - 'arn:aws:sns:'
+          - !Ref 'AWS::Region'
+          - ':'
+          - !Ref 'AWS::AccountId'
+          - ':marketplace-test-topic'
+
+  DevPortalLambdaFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ../lambdas/backend
+      Handler: index.handler
+      MemorySize: 1024
+      Role: !GetAtt DevPortalLambdaExecutionRole.Arn
+      Runtime: nodejs4.3
+      Timeout: 30
+      Environment:
+        Variables:
+          WEBSITE_BUCKET_NAME: !Ref DevPortalSiteS3BucketName
+
+      # Adds the API as a trigger
+      Events:
+        ProxyApiRoot:
+          Type: Api
+          Properties:
+            RestApiId: !Ref ApiGatewayApi
+            Path: /
+            Method: ANY
+        ProxyApiGreedy:
+          Type: Api
+          Properties:
+            RestApiId: !Ref ApiGatewayApi
+            Path: /{proxy+}
+            Method: ANY
+
+  SubscriptionListenerLambdaFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ../lambdas/listener
+      Handler: index.handler
+      MemorySize: 128
+      Role: !GetAtt DevPortalLambdaExecutionRole.Arn
+      Runtime: nodejs4.3
+      Timeout: 30
+
+  # PreSignupLambdaExecutionRole:
```

Comment:
Can we scrap the commented out bit?

Reply:
Yep. Right now, all the lambda functions are using the same policy. I'd love to split the policy out when I get a chance, but there's no reason to have this around for now.